### PR TITLE
Update `aea check-packages` by adding a check on PUBLIC_ID

### LINKED ISSUE #303: Improved consistency checks
**Is your feature request related to a problem? Please describe.**
[A clear and concise description of what the problem is. Ex. I'm always frustrated when [...]
](https://github.com/valory-xyz/open-autonomy/pull/1306#discussion_r965607665)

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.
](https://github.com/valory-xyz/open-autonomy/pull/1306#discussion_r965607665)

Could also check for third-party imports being declared as dependencies

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR adds one more check in `aea check-packages` by also checking the presence of the constant `PUBLIC_ID` for connections an skills.

## Fixes

Fix #303 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/cli/check_packages.py [hunk 1]
```diff
@@ -258,6 +309,30 @@ def check_author(configuration_file: Path, expected_author: str) -> None:
...
+    if package_type == PackageType.CONNECTION:
+        module_name_to_load = Path("connection.py")
+    elif package_type == PackageType.SKILL:
+        module_name_to_load = Path("__init__.py")
+    else:
+        # no check to do.
+        return
+    module_path_to_load = configuration_file.parent / module_name_to_load
+    content = module_path_to_load.read_text()
+    matches = re.findall("^PUBLIC_ID = (.*)", content, re.MULTILINE)
```
Comment (marcofavorito) [inline aea/cli/check_packages.py:327]: Tiny improvement: allow zero or more spaces.
```suggestion
    matches = re.findall("^PUBLIC_ID *= *(.*)", content, re.MULTILINE)
```

File: aea/cli/check_packages.py [hunk 2]
```diff
@@ -258,6 +309,30 @@ def check_author(configuration_file: Path, expected_author: str) -> None:
...
+    if len(matches) != 1:
+        raise PublicIdDefinitionError(package_type, expected_public_id, len(matches))
+
+    public_id_code = matches[0]
+    if str(expected_public_id) not in public_id_code:
+        raise WrongPublicIdError(package_type, expected_public_id, public_id_code)
```
Comment (marcofavorito) [inline aea/cli/check_packages.py:333]: just saying: a false negative would be the string:

```python
PUBLIC_ID = PublicId("author", "name", "version")
```

Anyway, by convention we can say you have to use `PublicId.from_str`. 

Surely safer than using `eval` :)

Comment (DavidMinarsch) [reply aea/cli/check_packages.py:333]: Yes I know - this is a bit hacky. I didn't feel like coming up with the comprehensive regex.

Comment (DavidMinarsch) [reply aea/cli/check_packages.py:333]: Maybe you can generalise it in the next PR on this @marcofavorito ?

Comment (marcofavorito) [reply aea/cli/check_packages.py:333]: Yes. we could cover the other most probable pattern (see above). Something like:
```
^PUBLIC_ID *= *PublicId\( *("[a-zA-Z0-9_]+") *, *("[a-zA-Z0-9_]+") *, *("[a-zA-Z0-9_]+") *\)
```

Comment (DavidMinarsch) [reply aea/cli/check_packages.py:333]: Yes I think these are the two only ways of instantiating the object. And then we impose this as the requirement.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**